cloudigrade/account/util.py

```diff
@@ -126,80 +135,134 @@ def generate_aws_ami_messages(instances_data, ami_list):
     return messages
 
 
-def _create_exchange_and_queue(queue_name):
+def _get_sqs_queue_url(queue_name):
     """
-    Create a Kombu message Exchange and Queue.
+    Get the SQS queue URL for the given queue name.
+
+    This has the side-effect on ensuring that the queue exists.
 
     Args:
-        queue_name (str): The target queue's name
+        queue_name (str): the name of the target SQS queue
 
     Returns:
-        tuple(kombu.Exchange, kombu.Queue)
+        str: the queue's URL.
 
     """
-    exchange = kombu.Exchange(
-        settings.QUEUE_EXCHANGE_NAME,
-        'direct',
-        durable=True
-    )
-    message_queue = kombu.Queue(
-        queue_name,
-        exchange=exchange,
-        routing_key=queue_name
+    sqs = boto3.client('sqs')
+    try:
+        return sqs.get_queue_url(QueueName=queue_name)['QueueUrl']
+    except ClientError as e:
+        if e.response['Error']['Code'].endswith('.NonExistentQueue'):
+            return sqs.create_queue(QueueName=queue_name)['QueueUrl']
+        raise
+
+
+def _sqs_wrap_message(message):
+    """
+    Wrap the message in a dict for SQS batch sending.
+
+    Args:
+        message (object): message to encode and wrap
+
+    Returns:
+        dict: structured entry for sending to send_message_batch
+
+    """
+    return {
+        'Id': str(uuid.uuid4()),
+        # Yes, the outgoing message uses MessageBody, not Body.
+        'MessageBody': base64.b64encode(
```

Comment:
What is the reason for Base64 encoding a JSON payload?

Reply:
Hmmm… good catch. I've gone through so many phases of experimenting on this replacement that I forgot exactly _why_ this was necessary. It seems to be an artifact from a previous attempted implementation, and after some local testing today, it looks to be unnecessary. I'll rip it out in a subsequent push.